syntax = "proto3";

package rag.v1;

option go_package = "Chimera-RAG/api/rag/v1;v1";

// å®šä¹‰ RAG æ ¸å¿ƒæœåŠ¡
service RagService {
  // 1. èŠå¤©æ¥å£ (æµå¼)
  // v0.4.0 å˜æ›´: è¯·æ±‚ä¸­éœ€è¦æºå¸¦ OrgID/KbID ä»¥è¿›è¡Œéš”ç¦»æ£€ç´¢
  rpc ChatStream (ChatRequest) returns (stream ChatReply) {}

  // 2. è§£æå¹¶å…¥åº“æ¥å£
  // v0.4.0 å˜æ›´: ä¸å†ä¼ æ–‡ä»¶æµï¼Œè€Œæ˜¯ä¼  MinIO è·¯å¾„ï¼›ä¸å†è¿”å›å‘é‡ï¼Œåªè¿”å›ç»Ÿè®¡ä¿¡æ¯
  rpc ParseAndIngest (ParseRequest) returns (ParseResponse) {}
}

// =========================================
// èŠå¤©ç›¸å…³æ¶ˆæ¯
// =========================================

message ChatRequest {
  string query = 1;
  string session_id = 2;

  // ğŸ”¥ v0.4.0 æ ¸å¿ƒæ–°å¢ï¼šå¤šç§Ÿæˆ·è¿‡æ»¤å™¨
  // å¦‚æœä¼ äº† kb_idï¼Œåªåœ¨è¿™ä¸ªçŸ¥è¯†åº“æœ
  // å¦‚æœæ²¡ä¼  kb_id ä½†ä¼ äº† org_idï¼Œåœ¨è¯¥ç»„ç»‡ä¸‹æ‰€æœ‰åº“æœ
  int64 org_id = 3;
  int64 kb_id = 4;

  // å†å²è®°å½• (å¯é€‰ï¼Œç®€åŒ–ç‰ˆ)
  repeated Message history = 5;
}

message Message {
  string role = 1; // "user" or "assistant"
  string content = 2;
}

message ChatReply {
  string answer_delta = 1; // æµå¼ç”Ÿæˆçš„æ–‡æœ¬ç‰‡æ®µ
  string source_docs = 2;  // å¼•ç”¨æ¥æº (JSON å­—ç¬¦ä¸²æˆ–æ ¼å¼åŒ–æ–‡æœ¬)
}

// =========================================
// è§£æç›¸å…³æ¶ˆæ¯ (ETL)
// =========================================

message ParseRequest {
  // ğŸ”¥ å˜æ›´ 1: ä¼  MinIO è·¯å¾„ï¼Œè€Œä¸æ˜¯æ–‡ä»¶å†…å®¹
  // ä¾‹å¦‚: "kbs/1/uuid.pdf"
  string storage_path = 1;

  string file_name = 2;

  // ğŸ”¥ å˜æ›´ 2: ä¼ é€’å…ƒæ•°æ®ï¼Œç”¨äº Python å†™å…¥ Qdrant Payload
  int64 kb_id = 3;
  int64 doc_id = 4;
  int64 org_id = 5; // å¦‚æœæ˜¯ä¸ªäººåº“ï¼Œè¿™é‡Œä¼  0
}

message ParseResponse {
  bool success = 1;
  string error_msg = 2;

  // ğŸ”¥ å˜æ›´ 3: åªè¿”å›ç»Ÿè®¡æ•°æ®ï¼Œä¸è¿”å›å‘é‡
  int32 chunk_count = 3; // åˆ‡äº†å¤šå°‘ç‰‡
  int32 page_count = 4;  // æ€»é¡µæ•° (Docling è§£æç»“æœ)
  int32 token_usage = 5; // æ¶ˆè€— Token æ•° (å¯é€‰)
}